pkg/base/k8s/options.go

```diff
@@ -0,0 +1,28 @@
+package k8s
+
+var DefaultMaxBackoffRetries = 0
+
+type KubeClientRequestOptions struct {
+	MaxBackoffRetries *int
+}
+
+func DefaultKubeClientRequestOptions() KubeClientRequestOptions{
+	return KubeClientRequestOptions{
+		MaxBackoffRetries: &DefaultMaxBackoffRetries,
+	}
+}
+
+func mergeKubeClientRequestOptions(opts ...*KubeClientRequestOptions) KubeClientRequestOptions {
```

Comment:
This line can be denoted for example as `func merge(opts ...*ClientOptions ) ClientOptions`, much readable IMO. The meaning of function can be implied from its arguments and return values.

Reply:
the functions are used to be named by the template: action + subject of action